### PR TITLE
Launch kata-agent without using nsenter

### LINKED ISSUE #109: Make kata-agent create a new network namespace for containers
In the current implementation, a `kata-agent` process runs in a fixed network namespace `podns` of a pod VM, and 
creates new containers in the `podns` network namespace.

https://github.com/confidential-containers/cloud-api-adaptor/blob/b5ecb837f63dab71abf2b22d9f4cd729ea132271/ibmcloud/image/files/etc/systemd/system/kata-agent.service#L5-L10

This work around is necessary, since `kata-shim` drops network namespace specification in the OCI runtime spec configuration of a newly created container, and containers created by `kata-agent` runs on the same network namespace where `kata-agent` runs. This just works, since the original `kata-agent` does not initiate any network activities.

https://github.com/kata-containers/kata-containers/blob/6d56cdb9acc9cd39165ab0f7380226d28104128a/src/runtime/virtcontainers/kata_agent.go#L943-L959
```
	// Disable network namespace since it is already handled on the host by
	// virtcontainers. The network is a complex part which cannot be simply
	// passed to the agent.
	// Every other namespaces's paths have to be emptied. This way, there
	// is no confusion from the agent, trying to find an existing namespace
	// on the guest.
	var tmpNamespaces []grpc.LinuxNamespace
	for _, ns := range grpcSpec.Linux.Namespaces {
		switch ns.Type {
		case string(specs.CgroupNamespace):
		case string(specs.NetworkNamespace):
		default:
			ns.Path = ""
			tmpNamespaces = append(tmpNamespaces, ns)
		}
	}
	grpcSpec.Linux.Namespaces = tmpNamespaces
```

However, there are some security and operational concerns with this approach, since the following network communication is done in the `podns`, and is possibly interfered by containers. 
- Image pulling on a pod VM (work around b5ecb837f63dab71abf2b22d9f4cd729ea132271)
- Abstract Unix domain socket communication between `kata-agent` and `agent-protocol-forwarder`

If `kata-agent` can create a new network namespace for containers, `kata-agent` and newly created containers run on separated network namespaces, and concerns mentioned above do not occur.

### PR DESCRIPTION
**Pull request by yoheiueda**
This PR makes kata-agent create containers in a specified network namespace. Without this patch, kata-agent creates containers in the host network namespace.

We no longer requires the `nsenter` workaround (#95), and cloud-api-adaptor becomes ready for transition to image.rs.

In addition to the above change, this PR add the following changes.

*  Add agent proto logging
*  Move the DNS workaround code to forwarder (#104)
*  Remove unused option (-host-interface)

Fixes #109

### CODE REVIEW

File: pkg/forwarder/interceptor/interceptor.go [hunk 1]
```diff
@@ -46,7 +53,73 @@ func NewInterceptor(agentSocket, nsPath string) Interceptor {
...
+}
+
+func (i *interceptor) CreateContainer(ctx context.Context, req *pb.CreateContainerRequest) (*types.Empty, error) {
+
+	logger.Printf("CreateContainer: containerID:%s", req.ContainerId)
+
+	res, err := i.Redirector.CreateContainer(ctx, req)
+
+	if err != nil {
+		logger.Printf("CreateContainer fails: %v", err)
```
Comment (bpradipt) [inline pkg/forwarder/interceptor/interceptor.go:62]: nit: wondering if replacing "fails" with "failed with error" be better ?

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @yoheiueda

File: pkg/forwarder/interceptor/interceptor.go (hunk 1 above)
Comment (yoheiueda) [reply pkg/forwarder/interceptor/interceptor.go:62]: OK. I'll change the message.

**Review by bpradipt — APPROVED** [review]
/lgtm

**Pull request merged**